server/data/data-api/src/main/java/org/apache/james/dlp/api/DLPRule.java

```diff
@@ -0,0 +1,184 @@
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.dlp.api;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+
+public class DLPRule {
+
+    public static class Builder {
+        public static final boolean NOT_TARGETTED = false;
+        private Optional<Boolean> targetsSender;
+        private Optional<Boolean> targetsRecipients;
+        private Optional<Boolean> targetsContent;
+        private Optional<String> explanation;
+        private Optional<String> expression;
+
+        public Builder() {
+            targetsSender = Optional.empty();
+            targetsRecipients = Optional.empty();
+            targetsContent = Optional.empty();
+            explanation = Optional.empty();
+            expression = Optional.empty();
+        }
+
+        public Builder targetsSender() {
+            this.targetsSender = Optional.of(true);
+            return this;
+        }
+
+        public Builder targetsRecipients() {
+            this.targetsRecipients = Optional.of(true);
+            return this;
+        }
+
+        public Builder targetsContent() {
+            this.targetsContent = Optional.of(true);
+            return this;
+        }
+
+        public Builder expression(String expression) {
+            this.expression = Optional.of(expression);
+            return this;
+        }
+
+        public Builder explanation(String explanation) {
+            this.explanation = Optional.of(explanation);
+            return this;
+        }
+
+        public DLPRule build() {
+            Preconditions.checkState(expression.isPresent(), "`expression` in mandatory");
+            return new DLPRule(
+                explanation,
+                expression.get(),
+                new Targets(
+                    targetsSender.orElse(NOT_TARGETTED),
+                    targetsRecipients.orElse(NOT_TARGETTED),
+                    targetsContent.orElse(NOT_TARGETTED)));
+        }
+    }
+
+    public static class Targets {
+        private final boolean senderTargeted;
+        private final boolean recipientTargeted;
+        private final boolean contentTargeted;
+
+        public Targets(boolean senderTargeted, boolean recipientTargeted, boolean contentTargeted) {
+            this.senderTargeted = senderTargeted;
```

Comment:
do we need to check if 3 targets are all `false` ?

Reply:
Why not. It would test explicitly default values and it's always better.

Reply:
that could be a not so much bad way to deactivate (temporarily) a rule.